File: examples/scala/wuserver.scala
//
//  Weather update server in Scala
//  Binds PUB socket to tcp://*:5556
//  Publishes random weather updates
//
//  author Giovanni Ruggiero
//  email giovanni.ruggiero@gmail.com

import java.util.Random
import org.zeromq.ZMQ

object wuserver {
  def main(args : Array[String]) {

		//  Prepare our context and publisher
		val context = ZMQ.context(1)
		val publisher = context.socket(ZMQ.PUB)

		publisher.bind("tcp://*:5556")

		//  Initialize random number generator
		val srandom = new Random(System.currentTimeMillis())
		while (true) {
			//  Get values that will fool the boss
			val zipcode: Integer = srandom.nextInt(100000) + 1
			val temperature: Integer = srandom.nextInt(215) - 80 + 1
			val relhumidity: Integer = srandom.nextInt(50) + 10 + 1

			//  Send message to all subscribers
			val update = String.format("%05d %d %d\u0000", zipcode, temperature, relhumidity);
			publisher.send(update.getBytes(), 0)
		}
	}
}

File: examples/OCaml/wuserver.ml
(**
  Weather update server
  Binds PUB socket to tcp://*:5556
  Publishes random weather updates
*)

open Zmq
open Helpers

let () =
  with_context @@ fun ctx ->
  with_socket ctx Socket.pub @@ fun pub ->
  Socket.bind pub "tcp://*:5556";
  Socket.bind pub "ipc://weather.ipc";
  Random.self_init ();
  while not !should_exit do
    (* Get values that will fool the boss *)
    let zipcode = Random.int 100_000 in
    let temperature = Random.int 215 - 80 in
    let relhumidity = Random.int 50 + 10 in
    (* Send message to all subscribers *)
    Socket.send pub (Printf.sprintf "%05d %d %d" zipcode temperature relhumidity);
  done
